### PR TITLE
Class-level resource access type

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
cc @dain @electrum

File: core/trino-main/src/main/java/io/trino/server/security/ResourceAccessType.java [hunk 1]
```diff
@@ -46,18 +46,14 @@ public AccessType getAccessType(ResourceInfo resourceInfo)
...
-            if (resourceInfo.getResourceClass().getDeclaringClass() != null) {
-                accessType = resourceAccessTypeLoader.getAccessType(resourceInfo.getResourceClass().getDeclaringClass());
+            // check if the resource class or it's parents has an access type declared for all methods
+            Class<?> current = resourceInfo.getResourceClass();
+            while (current != null) {
+                accessType = resourceAccessTypeLoader.getAccessType(current);
                 if (accessType.isPresent()) {
                     return accessType.get();
                 }
+                current = current.getDeclaringClass();
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/server/security/ResourceAccessType.java:56]: Reintroduce comment about it is for handlint nested classes.

**Review by losipiuk — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/server/security/ResourceAccessType.java (hunk 1 above)
Comment (wendigo) [reply core/trino-main/src/main/java/io/trino/server/security/ResourceAccessType.java:56]: Reintroduced

**Comment by mosabua** [comment]
I assume no release notes needed fyi..

**Comment by wendigo** [comment]
@mosabua not needed

**Pull request merged**